### PR TITLE
Add object

### LINKED ISSUE #26: Implement add art object activity
We need to have add object activity. Suggesting to style in the same way as instagram add picture. This activity need to be opened on floating button click.

### PR DESCRIPTION
**Pull request by sanovikov71**
Fix #26 

It's a partial implementation just for you to look through the code if you want to.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -86,7 +87,7 @@ dependencies {
 
     compile 'com.squareup.retrofit:retrofit:2.0.0-beta2'
     compile 'com.squareup.retrofit:converter-gson:2.0.0-beta2'
-    compile 'com.squareup.picasso:picasso:2.5.2'
+    compile 'com.github.bumptech.glide:glide:3.7.0'
```
Comment (genaforvena) [inline app/build.gradle:90]: why glide is better than picasso?

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt [hunk 2]
```diff
@@ -59,10 +60,33 @@ class DataSource(private val realm: Realm, private val handler: Handler) {
                 .map { ArtObjectUi(it as RealmArtObject) }
     }
 
+    fun addArtObject(newName: String, newAuthor: String, newImageUrl: Uri?, location: Location?) {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt]: location cannot be null i think

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import org.imozerov.streetartview.R
+import org.imozerov.streetartview.StreetArtViewApp
+import org.imozerov.streetartview.storage.DataSource
+import rx.subscriptions.CompositeSubscription
+import javax.inject.Inject
+
+/**
+ * Created by sergei on 18.02.16.
+ */
+class AddArtObjectActivity : AppCompatActivity(), GoogleApiClient.ConnectionCallbacks,
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:38]: We should think about implementing MVP here.

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                .addOnConnectionFailedListener({})
+                .addApi(LocationServices.API)
+                .build();
+
+    }
+
+    override fun onStart() {
+        googleApiClient?.connect();
+        compositeSubscription = CompositeSubscription()
+        super.onStart()
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:147]: I would expect it to be the first call in onStart callback.

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    public override fun onStop() {
+        googleApiClient?.disconnect();
+        compositeSubscription!!.unsubscribe()
+        super.onStop()
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:153]: same

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    fun openGalleryToSetImageForArtObjectWithId() {
```
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:156]: I would inline this method.

File: app/build.gradle (hunk 1 above)
Comment (sanovikov71) [inline app/build.gradle:90]: Image loading is faster and smoother

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt (hunk 2 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt]: should be corrected after location implementation

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt (hunk 3 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:38]: I'll consider it

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt (hunk 4 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:147]: hz

File: app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt (hunk 6 above)
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/add/AddArtObjectActivity.kt:156]: ok

File: app/build.gradle (hunk 1 above)
Comment (genaforvena) [inline app/build.gradle:90]: HUYA!

**Pull request closed without merging**